src/Scripting/CSharpTest/CommandLineRunnerTests.cs

```diff
@@ -566,6 +566,26 @@ public void RelativePath()
             }
         }
 
+        [Theory]
+        [InlineData(null, null)]
+        [InlineData("c:", null)]
+        [InlineData("c:\\", null)]
+        [InlineData("c:\\first", "c:\\")]
+        [InlineData("c:\\first\\", "c:\\first")]
+        [InlineData("c:\\first\\second", "c:\\first")]
+        [InlineData("c:\\first\\second\\", "c:\\first\\second")]
+        [InlineData("c:\\first\\second\\third", "c:\\first\\second")]
+        [InlineData("\\", null)]
+        [InlineData("\\first", "\\")]
+        [InlineData("\\first\\", "\\first")]
+        [InlineData("\\first\\second", "\\first")]
+        [InlineData("\\first\\second\\", "\\first\\second")]
+        [InlineData("\\first\\second\\third", "\\first\\second")]
+        public void TestGetDirectoryName_Windows(string path, string expectedOutput)
```

Comment:
>public [](start = 8, length = 6)

Do we need to test relative paths as well?

Reply:
Doesn't hurt. I'll add.
The main point of this test is to show when we're returning null

---
In reply to: [378586494](https://github.com/dotnet/roslyn/pull/41063#discussion_r378586494) [](ancestors = 378586494)